fortuna/prob_model/predictive/classification.py

```diff
@@ -424,3 +427,145 @@ def _log_lik_fun(outputs):
+            jnp.concatenate((train_data[1], test_data_grid[1]), axis=0),
+        )
+
+        train_test_data_loader = DataLoader.from_array_data(
+            train_test_data, batch_size=128, prefetch=True
+        )
+
+        # returns n_posterior_samples x (n_test + n_test +n)
+        ensemble_train_test_log_probs = self.ensemble_log_prob(
+            data_loader=train_test_data_loader, n_posterior_samples=n_posterior_samples
+        )
+
+        # Divide back out into test set grid and training
+        ensemble_train_log_probs = ensemble_train_test_log_probs[
+            :, 0:n
+        ]  # training log likelihood
+        ensemble_test0_log_probs = ensemble_train_test_log_probs[
+            :, n : n + n_test
+        ]  # test log likelihoods at Y = 0
+        ensemble_test1_log_probs = ensemble_train_test_log_probs[
+            :, n + n_test : n + 2 * n_test
+        ]  # test log likelihoods at Y = 1
+
+        # log likelihood values for each test input, posterior sample, and grid point (shape =  n_test x  n_posterior_samples x 2)
+        ensemble_testgrid_log_probs = jnp.concatenate(
+            (
+                ensemble_test0_log_probs.T.reshape(-1, n_posterior_samples, 1),
+                ensemble_test1_log_probs.T.reshape(-1, n_posterior_samples, 1),
+            ),
+            axis=2,
+        )
+
+        @jit  # compute rank of nonconformity score (unnormalized by n+1)
+        def _compute_cb_region_IS(ensemble_testgrid_log_probs):
+            n = jnp.shape(ensemble_train_log_probs)[
+                1
+            ]  # ensemble_train_log_probs is n_posterior_samples x n; training data log likelihood values
+            n_plot = jnp.shape(ensemble_testgrid_log_probs)[
+                1
+            ]  # ensemble_testgrid_log_probs is n_posterior_samples x n_plot_grid;  test data log likelihood values for each posterior sample
+            rank_cp = jnp.zeros(n_plot)
+
+            # compute importance sampling weights and normalizing
+            wjk = jnp.exp(ensemble_testgrid_log_probs.T)
+            Zjk = jnp.sum(wjk, axis=1).reshape(-1, 1)
+
+            # compute predictives for y_i,x_i and y_new,x_n+1
+            p_cp = jnp.dot(wjk / Zjk, jnp.exp(ensemble_train_log_probs))
+            p_new = jnp.sum(wjk**2, axis=1).reshape(-1, 1) / Zjk
+
+            # compute nonconformity score and sort
+            pred_tot = jnp.concatenate((p_cp, p_new), axis=1)
+            rank_cp = jnp.sum(pred_tot <= pred_tot[:, -1].reshape(-1, 1), axis=1)
+
+            # Compute region of grid which is in confidence set
+            region_true = rank_cp > error * (n + 1)
+            return region_true
+
+        # Compute CB interval for each
+        conf_set = vmap(_compute_cb_region_IS)(ensemble_testgrid_log_probs)
+
+        if ESS:
+            ## DIAGNOSE IMPORTANCE WEIGHTS ##
+            @jit  # compute Effective sample size
+            def _diagnose_is_weights(ensemble_testgrid_log_probs):
```

Comment:
For this method, I have the same comments as above: explicit IS, write `n_train`, condense `rank_cp`.